\documentclass{article}
% \VignettePackage{ASGSCA}
% \VignetteIndexEntry{Association Studies using Generalized Structured Equation Models.}
\usepackage{graphicx}
\usepackage[margin=2cm]{geometry}
\usepackage[colorlinks=true,urlcolor=blue]{hyperref}
\usepackage{array}
\usepackage{color}
\usepackage{underscore}
%\usepackage[authoryear,round]{natbib}
\bibliographystyle{plainnat}
\usepackage{verbatim}
%\usepackage[utf8]{inputenc} % for UTF-8/single quotes from sQuote()
\usepackage{Sweave}%
% for bold symbols in mathmode
\usepackage{bm}
\usepackage{setspace}
\doublespacing

\SweaveOpts{keep.source=TRUE}

\title{Using ASGSCA}
\author{Celia Greenwood, Stepan Grinek. }

\date{\today}
\sloppy
\hyphenpenalty 10000


\begin{document}
\SweaveOpts{concordance=TRUE}

\maketitle


%% Note: These are explained in '?RweaveLatex' :
%<<preliminaries, echo=FALSE>>=
%options(width=75)
%@

\section{Introduction}


The \textbf{newNorm} provides a function to normalize Illumina Infinium Human Methylation 450 BeadChip (Illumina 450K). This package is a further improvement of normalization method intoroduced by
\cite{Fortin2014, Aryee2014} recently. In addition to percentile specific adjustments in funNorm, newNorm also corrects for tissue or cel type, which results in significant improvement of performance \cite{newNorm}. Another advantage of the method is that it uses Partial Least Square regression (PLS) instead of Principal Component regression.

Min 2 celltypes

Curent version of package does not provide a special treatment for X chromosome, to be included in subsequent version of the package. 

\section{Package use}

User is expected to provide an annotation table, containing information on probe names, probe type, color (for probes of type I), two methylation matrices with signal A and B (unmethylated and methylated sygnals) and matrices with control probes signal, separately for red and green channel, list of types of control probes and list of cell types. Annotation file and matrices with A, B and control signals are expected to be the output of Genome Studio (http://www.illumina.com/applications/microarrays/microarray-software/genomestudio.html) or, alternatevily, extracted from IDAT files generated on Illumina array platforms. Package also contains default annotation and list of control probes, which are loaded if Annot and cp.types are set to NULL in the newnorm function call.   

We created small sample data set of $N=93$ individuals with 1000 probes to demonstrate the usage of our package. Samples are taken from cord blood and placenta. 
Here is a basic call to normalize sample data set: 
 %The code to be inserted
%\begin{Verbatim}
<<>>= 
    library(newNorm)
    data("data")
    ncmp <- 4
    newnormout <- newnorm(sigA=sigAsample, sigB=sigBsample, Annot=Annotsample, 
                      controlred=matred, controlgrn=matgrn, 
                      cp.types=cp.types, cell_type = cell_type,
                       ncmp=ncmp, save.quant=TRUE, save.loess=TRUE, apply.loess=TRUE, logit.quant=TRUE, validate=F)

@
%\end{Verbatim}
Number of PLS components, ncmp, could be chosen after examining of the graph generated by cross-validation procedure. The following call performs cross-validation to find optimal value of parameter ncmp for PLS regression trying numbers of components from 1 to 5: 

%\begin{Verbatim}
<<>>= 
    #This call  will perform cross validation of to find optimal value of parameter ncmp for PLS regression:
    newnormout <- newnorm(sigA=sigAsample, sigB=sigBsample,
                      controlred=matred, controlgrn=matgrn, 
                      cp.types=cp.types, cell_type = cell_type,
                      save.quant=TRUE, save.loess=TRUE, apply.loess=FALSE, logit.quant=TRUE, validate=5)

@
%\end{Verbatim}

Add graph and explanations of its meaning.

\begin{thebibliography}{}

\bibitem{deLeeuw1976} de Leeuw, J., Young, F. W., and Takane, Y. (1976). Additive structure in qualitative data: An alternating least squares method with optimal scaling features. \emph{Psychometrika}, 41, 471-503.
\bibitem{Fortin2014} Fortin, J.-P., et al. (2014). Functional normalization of 450K methylation array data improves replication in large cancer studies. \emph{Genome Biology},  15: p. 503.
\bibitem{Aryee2014} Aryee, M.J., et al. (2014). Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays. \emph{Bioinformatics}, 30(10): p. 1363-9.
\bibitem{newNorm}  Kathleen Oros Klein et al. (2015). newNorm: An improvement of the funNorm normalization method for methylation data from multiple cell or tissue types.  \emph{Bioinformatics}, ???. 


\end{thebibliography}
 




\end{document}